Firestore/Example/Tests/Util/FSTIntegrationTestCase.mm

```diff
@@ -135,16 +135,14 @@ - (FIRFirestore *)firestoreWithProjectID:(NSString *)projectID {
   FSTDispatchQueue *workerDispatchQueue = [FSTDispatchQueue
       queueWith:dispatch_queue_create("com.google.firebase.firestore", DISPATCH_QUEUE_SERIAL)];
 
-  FSTEmptyCredentialsProvider *credentialsProvider = [[FSTEmptyCredentialsProvider alloc] init];
-
   FIRSetLoggerLevel(FIRLoggerLevelDebug);
   // HACK: FIRFirestore expects a non-nil app, but for tests we cheat.
   FIRApp *app = nil;
   FIRFirestore *firestore = [[FIRFirestore alloc]
         initWithProjectID:projectID
                  database:util::WrapNSStringNoCopy(DatabaseId::kDefaultDatabaseId)
            persistenceKey:persistenceKey
-      credentialsProvider:credentialsProvider
+      credentialsProvider:new EmptyCredentialsProvider()  // passing ownership
```

Comment:
Since you're passing ownership, a unique_ptr would be a better alternative here. But I'm not sure how that interacts with obj-c++? (So this may not be possible...)

Ideally, this might read something like this:
```
credentailsProvider:std::move(new EmptyCredentialsProvider())
```

Further reading on the subject:
https://engdoc.corp.google.com/eng/doc/devguide/cpp/primer.shtml?cl=head#google_pointers will refer you to here:
https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/

Update: This issue seems to occur through this PR. In particular, there's a lot of comments that say either:
`// passing ownership` or `// not passing ownership`. I won't be able to keep that straight. :(  unique_ptr/move is the (modern) c++ way to handle this. If that doesn't work, could we investigate maybe just making a bunch of extra copies so that everything's consistent? (i.e. trade a few extra allocs in return for me staying sane.)

Reply:
FIRFirestore (and the C++ equivalent once it exists) will own the constellation of objects related to that instance. Therefore this parameter should be a `unique_ptr<CredentialsProvider>` because we're intending that FIRFirestore take ownership.

In general, even within internal APIs we shouldn't be passing ownership via raw pointers (and so comments like `// passing ownership` should not exist. All raw pointers should be non-owning pointers that we know have a lifetime that exceeds the duration that the callee might capture the parameter.